runtime/lua/vim/lsp/sync.lua

```diff
@@ -59,31 +58,14 @@ local function byte_to_utf(line, byte, offset_encoding)
   -- convert to 0 based indexing for str_utfindex
   byte = byte - 1
 
-  local utf_idx
-  local _
-  -- Convert the byte range to utf-{8,16,32} and convert 1-based (lua) indexing to 0-based
-  if offset_encoding == 'utf-16' then
-    _, utf_idx = str_utfindex(line, byte)
-  elseif offset_encoding == 'utf-32' then
-    utf_idx, _ = str_utfindex(line, byte)
-  else
-    utf_idx = byte
-  end
+  local utf_idx = vim.lsp.util.str_utfindex(line, byte, offset_encoding)
 
   -- convert to 1 based indexing
   return utf_idx + 1
 end
 
 local function compute_line_length(line, offset_encoding)
-  local length
-  local _
-  if offset_encoding == 'utf-16' then
-     _, length = str_utfindex(line)
-  elseif offset_encoding == 'utf-32' then
-    length, _ = str_utfindex(line)
-  else
-    length = #line
-  end
+  local length = vim.lsp.util.str_utfindex(line, nil, offset_encoding)
```

Comment:
Where is the handling of utf-8?

Reply:
nvm, I see what you've done. I would strongly prefer *not* eliding built-in API functions.

Reply:
See the definition of this new wrapper in `util.lua`

Reply:
see my comment above :)

Reply:
> I would strongly prefer not eliding built-in API functions.

I just don't think that we have a good alternative here yet if we want to avoid repeating this conditional block throughout the code.

Reply:
you can name it literally anything else.

Reply:
Oh gotcha. Will do. I thought you wanted to still use the old function. Turns out I don't know what "elide" means :P

Reply:
How about `lsp_utfindex` and `lsp_byteindex`?